libs/libarchfpga/src/physical_types_util.cpp

```diff
@@ -1061,18 +1081,98 @@ int get_class_num_from_pin_physical_num(t_physical_tile_type_ptr physical_tile,
     }
 }
 
-bool is_pin_on_tile(t_physical_tile_type_ptr physical_tile, int physical_num) {
-    return (physical_num < physical_tile->num_pins);
+bool is_primitive_pin(t_physical_tile_type_ptr physical_tile, int pin_physical_num) {
```

Comment:
For all the function implementations in this file, you should add some high level comments to make it more readable.